src/lib.rs

```diff
@@ -842,4 +843,12 @@ mod tests {
         let slice = bytes.as_slice_of::<u16>().unwrap();
         assert_eq!(slice, &[]);
     }
+
+    #[test]
+    fn unit() {
+        let slice: [(); 4] = [(), (), (), ()];
+        let bytes = slice.as_byte_slice();
```

Comment:
Does the other way around also work, going from some byte slice to a `&[()]`? Probably not in any meaningful way, or what would be the expected result slice length?

Currently it probably panics somewhere, doesn't it?